lib/mackerel/service.rb

```diff
@@ -0,0 +1,54 @@
+module Mackerel
+
+  class Service
+
+    attr_accessor :name, :memo, :roles
+
+    def initialize(args = {})
+      @name                            = args["name"]
+      @memo                            = args["memo"]
+      @roles                           = args["roles"]
+    end
+
+    def to_h
+      instance_variables.flat_map do |name|
+        respond_to?(name[1..-1]) ? [name[1..-1]] : []
+      end.each_with_object({}) do |name, hash|
+        hash[name] = public_send(name)
+      end.delete_if { |key, val| val == nil }
+    end
+
+    def to_json(options = nil)
+      return to_h.to_json(options)
+    end
+
+  end
+
+  module REST
+    module Service
+
+      def get_services()
+        command = ApiCommand.new(:get, '/api/v0/services')
+        command.headers['X-Api-Key'] = @api_key
+        data = command.execute(client)
+        data['services'].map {|s| Mackerel::Service.new(s) }
+      end
+
+      def get_roles(serviceName)
+        command = ApiCommand.new(:get, "/api/v0/services/#{serviceName}/roles")
+        command.headers['X-Api-Key'] = @api_key
+        data = command.execute(client)
+        data['roles'].map {|s| Mackerel::Role.new(s) }
+      end
+
+      def get_service_metric_names(serviceName)
```

Comment:
Are there any reasons to use `get_service_metric_names` rather than`get_metric_names`?

Reply:
I commented above, but now I realized my previous review was wrong..
Those all methods are expanded under `Mackerel::Client` class, therefore the method names should not be confusing with other methods.
So, to distinguish from `get_host_metric_names`, your first choice `get_service_metric_names` is better.

I'll change this method names in next p-r, so now I'd like to keep this method name as is in this p-r.